src/main/kotlin/com/team4099/robot2023/subsystems/vision/Vision.kt

```diff
@@ -1,18 +1,286 @@
 package com.team4099.robot2023.subsystems.vision
 
-import com.team4099.lib.vision.VisionMeasurement
+import com.team4099.lib.hal.Clock
+import com.team4099.lib.vision.PoseEstimator
 import com.team4099.robot2023.config.constants.FieldConstants
 import com.team4099.robot2023.config.constants.VisionConstants
+import com.team4099.robot2023.subsystems.vision.camera.CameraIO
+import com.team4099.robot2023.util.PolynomialRegression
+import edu.wpi.first.math.VecBuilder
 import edu.wpi.first.wpilibj2.command.SubsystemBase
 import org.littletonrobotics.junction.Logger
-import org.team4099.lib.apriltag.AprilTagFieldLayout
+import org.team4099.lib.geometry.Pose2d
 import org.team4099.lib.geometry.Pose3d
 import org.team4099.lib.geometry.Pose3dWPILIB
+import org.team4099.lib.geometry.Quaternion
+import org.team4099.lib.geometry.Rotation3d
+import org.team4099.lib.units.base.Time
+import org.team4099.lib.units.base.inMeters
+import org.team4099.lib.units.base.inMilliseconds
+import org.team4099.lib.units.base.inSeconds
+import org.team4099.lib.units.base.meters
 import org.team4099.lib.units.base.seconds
+import org.team4099.lib.units.derived.radians
+import java.util.function.Consumer
+import java.util.function.Supplier
 
-class Vision(cameras: VisionIO) : SubsystemBase() {
-  val ambiguityThreshold = 0.15
-  val targetLogTime = 0.05.seconds
-  val cameraPoses = VisionConstants.CAMERA_TRANSFORMS
+class Vision(vararg cameras: CameraIO) : SubsystemBase() {
+  val io: List<CameraIO> = cameras.toList()
+  val inputs = List(io.size) { CameraIO.CameraInputs() }
 
+  companion object {
+    val ambiguityThreshold = 0.15
+    val targetLogTime = 0.05.seconds
+    val cameraPoses = VisionConstants.CAMERA_TRANSFORMS
+
+    val xyStdDevModel =
```

Comment:
I know this is how 6328 did it, but I feel like these should be per-camera. This is highly dependent on camera resolution etc, so if you (for example) added a limelight or something as a supplementary camera, it would need different standard deviations